gensim/models/experimental/drmm_tks.py

```diff
@@ -0,0 +1,726 @@
+        epochs : int
+            The number of epochs for which the model should train on the data
+        unk_handle_method : {'zero', 'random'}
+            The method for handling unkown words
+            'zero' : unknown words are given a zero vector
+            'random' : unknown words are given a uniformly random vector bassed on the word string hash.
+        validation_data: list of the form [test_queries, test_docs, test_labels]
+            where test_queries, test_docs  and test_labels are of the same form as
+            their counter parts stated above.
+        topk : int
+            the k topmost values in the interaction matrix between the queries and the docs
+        target_mode : {'ranking', 'classification'}
+            the way the model should be trained, either to rank or classify
+        """
+        self.queries = queries
+        self.docs = docs
+        self.labels = labels
+        self.word_counter = Counter()
+        self.text_maxlen = text_maxlen
+        self.topk = topk
+        self.word_embedding = word_embedding
+        self.word2index, self.index2word = {}, {}
+        self.normalize_embeddings = normalize_embeddings
+        self.model = None
+        self.epochs = epochs
+        self.validation_data = validation_data
+        self.target_mode = target_mode
+
+        if self.target_mode not in ['ranking', 'classification']:
+            raise ValueError("Unkown target_mode %s. It must be either 'ranking' or 'classification'" %
+                             self.target_mode)
+
+        if unk_handle_method not in ['random', 'zero']:
+            raise ValueError("Unkown token handling method %s" %
+                             str(unk_handle_method))
+        self.unk_handle_method = unk_handle_method
+
+        if self.queries is not None and self.docs is not None and self.labels is not None:
+            self.build_vocab()
+            self.train()
+        else:
+            logger.info("Vocab won't be built and Model won't be trained"
+                        " as data is either not provided or is incomplete.")
+
+    def build_vocab(self):
+        """Indexes all the words and makes an embedding_matrix which
+        can be fed directly into an Embedding layer"""
+
+        logger.info("Starting Vocab Build")
+
+        # get all the vocab words
+        for q in self.queries:
+            self.word_counter.update(q)
+        for doc in self.docs:
+            for d in doc:
+                self.word_counter.update(d)
+        for i, word in enumerate(self.word_counter.keys()):
+            self.word2index[word] = i
+            self.index2word[i] = word
+
+        self.vocab_size = len(self.word2index)
+        logger.info("Vocab Build Complete")
+        logger.info("Vocab Size is %d" % self.vocab_size)
+
+        logger.info("Building embedding index using pretrained word embeddings")
+        if type(self.word_embedding) == str:
```

Comment:
oh, but if I have non-glove txt embeddings, what I should to do? Please fully remove this, accept only keyedvectors always.